src/webots/core/WbLog.hpp

```diff
@@ -33,16 +33,35 @@ class WbLog : public QObject {
   Q_OBJECT
 
 public:
-  enum Level { DEBUG, INFO, WARNING, ERROR, FATAL, STATUS, STDOUT, STDERR };
+  enum Level { DEBUG, INFO, WARNING, ERROR, FATAL, STATUS, STDOUT, STDERR, LEVEL_SIZE };
+  enum Filter {
+    ALL,
+    WEBOTS,
+    ALL_WEBOTS,
+    CONTROLLERS,
+    ALL_CONTROLLERS,
+    WEBOTS_OTHERS,
+    ODE,
+    JAVASCRIPT,
+    PARSING,
+    COMPILATION,
+    FILTER_SIZE
+  };
 
   static WbLog *instance();
 
   // Webots messages, e.g. parse error
   // the 'message' argument should not be '\n' terminated
-  static void debug(const QString &message, bool popup = false);
-  static void info(const QString &message, bool popup = false);
-  static void warning(const QString &message, bool popup = false);
-  static void error(const QString &message, bool popup = false);
+  static void debug(const QString &message, bool popup = false, const QString &name = QString());
```

Comment:
Why passing a `QString` for the `name`? Passing a `WbLog::Filter` type would be simpler, no?
If for some reason a `QString` is really needed, we could add helper methods like:
```c++
   static void debug(const QString &message, bool popup = false, Filter filter);
```

Reply:
Because having the information about the type of filter is not enough, for example if the type is `WbLog::CONTROLLERS`, we have to know which controller it is too.